crates/router/src/core/payments/transformers.rs

```diff
@@ -1809,6 +1816,22 @@ impl<F: Clone> TryFrom<PaymentAdditionalData<'_, F>> for types::PaymentsAuthoriz
         ));
 
         // payment_method_data is not required during recurring mandate payment, in such case keep default PaymentMethodData as MandatePayment
+        let additional_payment_method_data = if payment_data.mandate_id.is_some() {
+            let parsed_additional_payment_data: Option<api_models::payments::AdditionalPaymentData> =
+                payment_data.payment_attempt
+                    .payment_method_data
+                    .clone()
+                    .and_then(|data| match data {
+                        serde_json::Value::Null => None, // Handle null case
+                        _ => Some(data.parse_value("AdditionalPaymentData")),
```

Comment:
Do we explicitly need to handle the `serde_json::Value::Null` case? Won't `parse_value()` handle that: if the JSON value is not the expected type (say a map in this case), then deserialization should fail?

Reply:
Not really , That was redundant

Reply:
Yes deserialization will fail